student.rb

```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom)
```

Comment:
*  Kindly add the necessary parameters to show that your `Student` class extends the `Parent` class as stated in the [project requirement ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Constructor%20extends%20parent%27s%20constructor%20by%20adding%20%40classroom%20and%20a%20parameter%20for%20it.) as all these parameters are needed to show inheritance from a person 
`
class Student < Person 
`
`
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
`
`
    super(age, name, parent_permission)
`
`
    @classroom = classroom
`
`
  end
`

Reply:
@Davidosky007 thank you for your review and for highlighting the points. I have corrected based on your good suggestion code above.

If I change the boolean parameter to have it like `parent_permission = true` there is a linter error raising and suggesting to have a boolean value like `parent_permission: true`. In this case,  the parameters passing to `super` are only accepting either 1 or 2. like instead of `super(age, name, parent_permission)` ->`super(age, name)` so the final code is like this

```
require './person'

class Student < Person
  def initialize(age, classroom, name = "Unknown", parent_permission: true)
    super(age, name)
    @classroom = classroom
  end

  def play_hooky
    '¯(ツ)/¯'
  end
end
```

Shall I proceed like this? Please suggest.
Thank You for your time.

Reply:
> @Davidosky007 thank you for your review and for highlighting the points. I have corrected based on your good suggestion code above.
> 
> If I change the boolean parameter to have it like `parent_permission = true` there is a linter error raising and suggesting to have a boolean value like `parent_permission: true`. In this case, the parameters passing to `super` are only accepting either 1 or 2. like instead of `super(age, name, parent_permission)` ->`super(age, name)` so the final code is like this
> 
> ```
> require './person'
> 
> class Student < Person
>   def initialize(age, classroom, name = "Unknown", parent_permission: true)
>     super(age, name)
>     @classroom = classroom
>   end
> 
>   def play_hooky
>     '¯(ツ)/¯'
>   end
> end
> ```
> 
> Shall I proceed like this? Please suggest. Thank You for your time.

Yes @PRATAP-KUMAR since the parent_permission as been set to default true

Reply:
@Davidosky007 Thank you very much for your quick response. I did not expect such a quick reply. Thanks a lot.

Reply:
sorry to interrupt you again @Davidosky007 I am getting a linter error saying `Lint/UnusedMethodArgument` does this mean the default parameter can be avoided in the initialize line? like instead of `def initialize(age, classroom, name = "Unknown", parent_permission: true)` -> `def initialize(age, classroom, name = "Unknown")`

![image](https://user-images.githubusercontent.com/40719899/198971045-7644fabf-1b7a-4e82-a801-169514d3f296.png)

This one works without linter errors

```
require './person'

class Student < Person
  def initialize(age, classroom, name = 'Unknown')
    super(age, name)
    @classroom = classroom
  end

  def play_hooky
    '¯(ツ)/¯'
  end
end
```

Please suggest. Can I proceed with the above code?
Thanks for you valuable time and sorry If I am taking your time much.

Reply:
This is the terminal output showing Linter Error.

```
Offenses:

teacher.rb:4:57: W: Lint/UnusedMethodArgument: Unused method argument - parent_permission.
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
                                                        ^^^^^^^^^^^^^^^^^

3 files inspected, 1 offense detected
```